### PR TITLE
feat(node): end the support of Node.js versions 10, 11, 12, 13

### PR DESCRIPTION
**Pull request by arnaudbesnier**
BREAKING CHANGE: users will have to upgrade to Node.js version 14+ to ensure a full compatibility

### General

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by arnaudbesnier** [comment]
Screenshot 2023-05-24 at 13 59 58

Here is a first version of the error message, I’ll try to improve the display.

**Comment by arnaudbesnier** [comment]
Screenshot 2023-05-24 at 15 37 52

**Review by jeffladiray — COMMENTED** [review]
Should we temporarily disable required checks on node 10 on CI for the beta branch?

(as we will also need, once merge, to add 14/16/18/20 as required on beta)

File: src/hooks/init/check-compatible-node-version.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  try {
+    const nodeVersion = Number(process.version.split('.')[0].split('v')[1]);
+    if (nodeVersion < NODE_VERSION_MINIMUM) {
+      logger.log(
+        chalk.red(
+          `The Forest Admin toolbelt is not compatible with your current Node.js version (v${nodeVersion} detected). Please use Node.js v${NODE_VERSION_MINIMUM}+.`,
+        ),
+      );
+      process.exit(0);
```
[the remark this answers is not in the document]
Comment (arnaudbesnier) [reply src/hooks/init/check-compatible-node-version.ts:22]: I was wondering what was the right value to set here.
I was telling to myself this is expected behavior so we should set 0 in the sense it is not an unexpected error.
But I might be wrong.

**Comment by arnaudbesnier** [comment]
> Should we temporarily disable required checks on node 10 on CI for the beta branch?
> 
> (as we will also need, once merge, to add 14/16/18/20 as required on beta)

Done @jeffladiray

File: src/hooks/init/check-compatible-node-version.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffladiray) [reply src/hooks/init/check-compatible-node-version.ts:22]: `cat a_file_that_exist.txt && echo $?` returns 0
`cat a_file_that_doesn_t_exist.txt && echo $?` returns 1.

From my very far student knowledge, process exit code don't care about expected vs unexpected errors, but more about correct process termination (0) vs abnormal termination (>0).

Currently, all the process.exit code are 1 when an error is raised (Expected and Unexpected) so customers **may** have scripts that rely on this behavior?

**Review by jeffladiray — APPROVED** [review]

**Review by DrRaider — APPROVED** [review]

**Review by SteveBunlon — APPROVED** [review]
🚀

**Pull request merged**